drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,787 @@
+#define RV8263_BM_TE_ENABLE                 (0x01 << 2)
+#define RV8263_BM_TIE_ENABLE                (0x01 << 1)
+#define RV8263_BM_TI_TP_PULSE               (0x01 << 0)
+#define RV8263_BM_OS                        (0x01 << 7)
+#define RV8263C8_BM_SOFTWARE_RESET          (0x58)
+#define RV8263C8_BM_REGISTER_OFFSET         0x7F
+#define RV8263_YEAR_OFFSET                  (2000 - 1900)
+
+#define SECONDS_BITS  GENMASK(6, 0)
+#define MINUTES_BITS  GENMASK(7, 0)
+#define HOURS_BITS    GENMASK(5, 0)
+#define DATE_BITS     GENMASK(5, 0)
+#define MONTHS_BITS   GENMASK(4, 0)
+#define WEEKDAY_BITS  GENMASK(2, 0)
+#define YEAR_BITS     GENMASK(7, 0)
+#define VALIDATE_24HR BIT(6)
+
+#define MIN_SEC       0
+#define MAX_SEC       59
+#define MIN_MIN       0
+#define MAX_MIN       59
+#define MIN_HOUR      0
+#define MAX_HOUR      23
+#define MAX_WDAY      7
+#define MIN_WDAY      1
+#define MAX_MDAY      31
+#define MIN_MDAY      1
+#define MAX_MON       12
+#define MIN_MON       1
+#define MIN_YEAR_DIFF 0
+#define MAX_YEAR_DIFF 99
+
+#define DT_DRV_COMPAT microcrystal_rv_8263_c8
+
+LOG_MODULE_REGISTER(microcrystal_rv8263c8, CONFIG_RTC_LOG_LEVEL);
+
+struct rv8263c8_config {
+	struct i2c_dt_spec i2c_bus;
+	uint32_t clkout;
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	struct gpio_dt_spec int_gpio;
+#endif
+};
+
+struct rv8263c8_data {
+	struct k_sem lock;
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	const struct device *dev;
+#endif
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	struct gpio_callback gpio_cb;
+#endif
+
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	rtc_alarm_callback alarm_cb;
+	void *alarm_cb_data;
+	struct k_work alarm_work;
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	rtc_update_callback update_cb;
+	void *update_cb_data;
+	struct k_work update_work;
+#endif
+};
+
+static int rv8263c8_update_disable_timer(const struct device *dev)
+{
+	int err;
+	uint8_t reg;
+	const struct rv8263c8_config *config = dev->config;
+
+	/* Value 0 disables the timer. */
+	reg = 0;
+	err = i2c_burst_write_dt(&config->i2c_bus, RV8263C8_REGISTER_TIMER_VALUE, &reg,
+				 sizeof(reg));
+	if (err < 0) {
+		return err;
+	}
+
+	return i2c_burst_write_dt(&config->i2c_bus, RV8263C8_REGISTER_TIMER_MODE, &reg,
+				  sizeof(reg));
+}
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+static void rv8263c8_gpio_callback_handler(const struct device *p_port, struct gpio_callback *p_cb,
+					   gpio_port_pins_t pins)
+{
+	ARG_UNUSED(pins);
+	ARG_UNUSED(p_port);
+
+	struct rv8263c8_data *data = CONTAINER_OF(p_cb, struct rv8263c8_data, gpio_cb);
+
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	k_work_submit(&data->alarm_work);
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
```

Comment:
```suggestion
#if CONFIG_RTC_UPDATE
```

Reply:
@bjarki-trackunit how should we do it now? Both variants are not correct now. Can you please help me to solve this issue :)?